src/fheroes2/battle/battle_interface.cpp

```diff
@@ -6333,16 +6334,21 @@ void Battle::Interface::InterruptAutoBattleIfRequested( LocalEvent & le )
         return;
     }
 
+    // Identify which color requested the auto battle interrupt.
+    int color = arena.GetCurrentColor();
+    if ( arena.GetCurrentForce().GetControl() & CONTROL_AI ) {
+        color = arena.GetOppositeColor( color );
+    }
+
+    // The battle interruption is already scheduled, no need for the dialog.
+    if (color == _interruptAutoBattleForColor) {
+        return;
+    }
+
```

Comment:
Hi @dood-apo I would also add something like the following code after this check, because, in theory, there can be either 0 or the current color, the other options are unacceptable, because they should have already been processed before switching to this unit. If this is not the case, then there is a mistake in the logic somewhere.

```suggestion

    assert( _interruptAutoBattleForColor == 0 );

```